packages/babel-traverse/src/index.js

```diff
@@ -33,6 +33,7 @@ traverse.explode = visitors.explode;
 traverse.NodePath = require("./path");
 traverse.Scope    = require("./scope");
 traverse.Hub      = require("./hub");
+traverse.cache    = require("./cache");
```

Comment:
Yes it does and WeakMap support exists in most modern environments now.
To put things in context, between putting things on the AST and using a WeakMap, putting on the AST is objectively more leaky.

An example of leakiness that bit me hard and I'm sure will continue to. People (reasonably) do the following in transforms:

```
const SOME_NODE_I_WILL_REUSE = t.identifier('blah');
```

Then `SOME_NODE_I_WILL_REUSE` will travel the universe being used across different files at different times all the while carrying an outdated `NodePath` and `Scope` information.